src/components/SearchFilter.js

```diff
@@ -0,0 +1,100 @@
+import React from "react";
+import { connect } from "react-redux";
+import common from "../reducers/common";
+
+const filters = {
+  today: false,
+  recent: false,
+  favorited: false,
+  popular: false
+};
+
+const popularTags = {
+  reactjs: false,
+  javascript: false,
+  nodejs: false,
+  java: false,
+  go: false
+};
+
+class SearchFilter extends React.Component {
+  constructor(props) {
+    super(props);
+    debugger;
+    this.articles = props.articles;
+    console.log("articles in Search Filter: " + JSON.stringify(this.articles));
+
+    this.toggleFilters = value => {
+      if (filters[value] == false) {
+        filters[value] = true;
+      } else if (filters[value] == true) {
+        filters[value] = false;
+      }
+    };
+
+    this.filterArticles = () => {
+      var filteredArticles = this.articles;
```

Comment:
`var` is not recommended, either `let` or `const` is preferred (ideally always `const`)